elements/nuxeo-audit/nuxeo-audit-search.js

```diff
@@ -0,0 +1,238 @@
+              class="item"
+              role="widget"
+              label="[[i18n('documentHistory.filter.eventCategory')]]"
+              directory-name="eventCategories"
+              value="{{selectedEventCategory}}"
+              placeholder="[[i18n('documentHistory.filter.selectEventCategory')]]"
+              min-chars="0"
+            >
+            </nuxeo-directory-suggestion>
+          </div>
+        </template>
+      </nuxeo-card>
+      <nuxeo-card>
+        <nuxeo-data-table id="table" paginable nx-provider="provider" empty-label="[[i18n('documentHistory.empty')]]">
+          <nuxeo-data-table-column name="[[i18n('documentHistory.performedAction')]]" sort-by="eventId">
+            <template>[[formati18n('activity.', item.eventId)]]</template>
+          </nuxeo-data-table-column>
+          <nuxeo-data-table-column name="[[i18n('documentHistory.date')]]" sort-by="eventDate">
+            <template><nuxeo-date datetime="[[item.eventDate]]"></nuxeo-date></template>
+          </nuxeo-data-table-column>
+          <nuxeo-data-table-column name="[[i18n('documentHistory.username')]]" sort-by="principalName">
+            <template><nuxeo-user-tag user="[[item.principalName]]"></nuxeo-user-tag></template>
+          </nuxeo-data-table-column>
+          <nuxeo-data-table-column name="[[i18n('documentHistory.category')]]" sort-by="category">
+            <template>[[formati18n('activity.', item.category)]]</template>
+          </nuxeo-data-table-column>
+          <nuxeo-data-table-column name="[[i18n('documentHistory.comment')]]">
+            <template>
+              <a href$="[[_parseComment(item.comment)]]">[[_formatComment(item.comment)]]</a>
+            </template>
+          </nuxeo-data-table-column>
+          <nuxeo-data-table-column name="[[i18n('documentHistory.state')]]">
+            <template><nuxeo-tag uppercase>[[formatLifecycleState(item.docLifeCycle)]]</nuxeo-tag></template>
+          </nuxeo-data-table-column>
+        </nuxeo-data-table>
+      </nuxeo-card>
+    `;
+  }
+
+  static get is() {
+    return 'nuxeo-audit-search';
+  }
+
+  static get properties() {
+    return {
+      visible: {
+        type: Boolean,
+        value: false,
+        observer: '_refresh',
+      },
+
+      document: Object,
+
+      providerName: {
+        type: String,
+        value: '',
+      },
+
+      principalName: {
+        type: String,
+        value: '',
+      },
+
+      startDate: {
+        type: String,
+        notify: true,
+      },
+      endDate: {
+        type: String,
+        notify: true,
+      },
+      selectedEventTypes: {
+        type: Array,
+        value: [],
+      },
+      selectedEventCategory: {
+        type: String,
+        value: '',
+      },
+    };
+  }
+
+  static get observers() {
+    return ['_refresh(startDate, endDate, selectedEventTypes.*, selectedEventCategory, principalName)'];
+  }
+
+  getDocumentId() {
+    return this.document ? this.document.uid : '';
+  }
```

Comment:
Consider using a getter:
```suggestion
  get documentId() {
    return this.document ? this.document.uid : '';
  }
```
usage: `this.documentId`